File: front-end/src/app/shared/dtos/solicitacao-cliente-detalhe-dto.ts
import { EstadoSolicitacao } from "../models/solicitacao.model";

export interface SolicitacaoDetalheDTO {
  id: number;
  createdAt: string;
  clienteNome: string | null;
  clienteEmail: string | null;
  descricaoProduto: string;
  defeito?: string | null;
  manutencaoDescricao: string;
  manutencaoOrientacoes: string;
  estado: EstadoSolicitacao | string;
  orcamentoValor?: number | null;
  historico: Array<{
    quando: string;
    de: EstadoSolicitacao | string | null;
    para: EstadoSolicitacao | string;
    funcionario?: string | null;
  }>;
}

File: back-end/src/main/java/com/example/back_end/dtos/response/SolicitacaoClienteDetalheDTO.java
package com.example.back_end.dtos.response;

import java.math.BigDecimal;
import java.util.List;

import com.example.back_end.enums.EstadoSolicitacao;

public record SolicitacaoClienteDetalheDTO(
    Long id,
    String createdAt,
    String clienteNome,
    String clienteEmail,
    String descricaoProduto,
    String defeito,
    String manutencaoDescricao,
    String manutencaoOrientacoes,
    EstadoSolicitacao estado,
    BigDecimal orcamentoValor,
    List<HistoricoDTO> historico
) {}